Orange/preprocess/score.py

```diff
@@ -390,6 +393,86 @@ def score_data(self, data, feature):
         return weights
 
 
+class UnsupervisedScorer(Scorer):
+    """
+    Simple unsupervised scorer for datasets without target variable.
+    """
+    feature_type = Variable
+
+    def __call__(self, data, feature=None):
+        if feature is not None:
+            f = data.domain[feature]
+            data = data.transform(Domain([f], data.domain.class_vars))
+
+        for pp in self.preprocessors:
+            data = pp(data)
+
+        for var in data.domain.attributes:
+            if not isinstance(var, self.feature_type):
+                raise ValueError(
+                    "{} cannot score {} variables."
+                    .format(self.friendly_name,
+                            self._friendly_vartype_name(type(var))))
+
+        return self.score_data(data, feature)
+
+
+class MeanScorer(UnsupervisedScorer):
+    """
+    Simple scorer returning mean of the features.
+    Return NA for non-continuos columns to enable mixtures of discrete and continuous variables.
+    """
+    supports_sparse_data = True
+    friendly_name = "Mean"
+
+    def score_data(self, data, feature):
+        weights = np.nan + np.zeros((len(data.domain.attributes)))
```

Comment:
NANs are currently returned for discrete features. This is in theory correct, but messes up sorting in the rank widget.

If -inf is returned, it messes up the graphical bars. 

If zeros are returned, this is misleading, especially for VarianceScorer and DispersionScorer.

Any suggestions?